workflow_factory.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/microsoft/BladeMonRT/nodes"
+	"github.com/microsoft/BladeMonRT/workflows"
+)
+
+/** Class for parsing workflow definitions. */
+type WorkflowFactory struct {
+	nameToWorkflow map[string]WorkflowDescription
+}
+
+/** Class for the workflow description in the JSON. */
+type WorkflowDescription struct {
+	Type string `json:"type"`
+	Nodes []NodeDescription `json:"nodes"`
+}
+
+/** Class for the node description in the JSON. */
+type NodeDescription struct {
+	Name string `json:"name"`
+	Metadata interface{} `json:"metadata"`
+	Args interface{} `json:"args"`
+}
+
+func newWorkflowFactory(workflowsJson []byte) WorkflowFactory {
```

Comment:
Make this capitalized as it is a public function, make this change for `constructWorkflow` to

Reply:
It's very specific to GO but capitalization at the start of a function name means that we want the function to be exported so it can be used in other packages. Since 'newWorkflowFactory' is only used within this package, lowercase n is fine here I think.

Reply:
It's very specific to GO but capitalization at the start of a function name means that we want the function to be exported so it can be used in other packages. Since 'newWorkflowFactory' is only used within this package, lowercase n is fine here I think.